src/Model/CallbackIterator.php

```diff
@@ -17,44 +17,54 @@
 
 namespace MongoDB\Model;
 
-use Closure;
 use Iterator;
 use IteratorIterator;
 use ReturnTypeWillChange;
 use Traversable;
 
+use function call_user_func;
+
 /**
  * Iterator to apply a callback before returning an element
  *
  * @internal
+ *
+ * @template TKey
+ * @template TValue
+ * @template TCallbackValue
+ * @template-implements Iterator<TKey, TCallbackValue>
  */
 class CallbackIterator implements Iterator
 {
-    /** @var Closure */
-    private $callback;
+    /** @var callable(TValue, TKey): TCallbackValue */
+    private $callable;
```

Comment:
The property name can stay `callback`: `callable` is a type, not a purpose.

Reply:
Good point, updated!